### PR TITLE
implemented activity details

### PR DESCRIPTION
**Pull request by aminetunc**

### CODE REVIEW

File: components/ActivityCard.js [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+  const router = useRouter();
+
+  const handleClick = () => {
+    router.push(`/activity/${id}`);
+  };
 
-export default function ActivityCard({ title, categories, imageUrl }) {
   return (
-    <StyledCardSection>
+    <StyledCardSection onClick={handleClick}>
```
Comment (jan-riemenschneider) [inline components/ActivityCard.js:13]: If the entire card is intended as a link, bookmarking a specific activity could become challenging, especially as this button will likely be on the card later and may trigger a redirect when clicked.
Possible solutions: Use the activity's name or the activity image as a link.

**Review by jan-riemenschneider — COMMENTED** [review]
Your code looks great, there are only some small remarks. 
Check the comments

File: components/ActivityCard.js [hunk 2]
```diff
@@ -1,34 +1,46 @@
...
+import { useRouter } from "next/router";
 
-export default function ActivityCard({ title, categories, imageUrl }) {
-  return (
-    <StyledCardSection>
-      <StyledImageDiv>
+export default function ActivityCard({ id, title, categories, imageUrl }) {
+  const router = useRouter();
+
+  const handleClick = () => {
```
Comment (jan-riemenschneider) [inline components/ActivityCard.js:9]: Please remove your handleClick function as you're not using it anymore (you are now using Link)

File: components/ActivityCard.js [hunk 3]
```diff
@@ -1,34 +1,46 @@
...
-      <StyledTitle>{title}</StyledTitle>
-    </StyledCardSection>
-  );
+    </StyledImageContainer>
+    <StyledList>
+      {categories.map((category) => (
+        <StyledListItem key={category}>{category}</StyledListItem>
+      ))}
+    </StyledList>
+    <StyledTitle>
```
Comment (jan-riemenschneider) [inline components/ActivityCard.js:29]: disclaimer: this is just a question from our side. Wouldn't you rather wrap the Link around the title and not the other way around ?

Comment (aminetunc) [reply components/ActivityCard.js:27]: We took another look and made some adjustments based on your feedback. Thank you for your input!

File: components/ActivityCard.js [hunk 4]
```diff
@@ -1,34 +1,43 @@
...
+    </StyledImageContainer>
+    <StyledList>
+      {categories.map((category) => (
+        <StyledListItem key={category}>{category}</StyledListItem>
+      ))}
+    </StyledList>
+    <Link href={`/activity/${id}`}>
+    <StyledTitle>
+      {title}
+    </StyledTitle></Link>
```
Comment (Roland-Hufnagel) [inline components/ActivityCard.js:29]: That's fine but now the Link is as wide as the card and not as wide as the title. You can solve this by making the title `display: inline;`.

File: components/ActivityDetails.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+import Link from "next/link";
+import Image from "next/image";
+
+export default function ActivityDetails() {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const activity = activities.find(activity => activity.id === id);
+
+  if (!activity) return <p>Loading...</p>;
```
Comment (Roland-Hufnagel) [inline components/ActivityDetails.js:13]: The component shouldn't be responsible for checking the current route. It's a good practice to keep components as dumb as possible. Instead move the router logic to the page `pages/activity/[id].js` and pass the found activity to the component as a prop.

**Review by Roland-Hufnagel — CHANGES_REQUESTED** [review]
This looks great 💪. Only two small things. Please look at my comments and improve the code.

**Review by Roland-Hufnagel — APPROVED** [review]
Perfect 🚀

**Pull request merged**